charts/shoot-core/components/charts/kube-proxy/templates/kube-proxy-daemonset.yaml

```diff
@@ -45,6 +45,28 @@ spec:
           /sbin/sysctl -w net.ipv6.conf.all.disable_ipv6=1 || echo "IPv6 is already disabled. Doing nothing..."
         securityContext:
           privileged: true
+{{- end }}
+{{- if and .Values.enableCleanup (ne .Values.previousMode "") (semverCompare ">= 1.16" .Values.kubernetesVersion) }}
```

Comment:
Doing it this way, every time the `kube-proxy` pod is restarted it will clean the previous rules.
This might cause service disruptions for a short period of time.

Reply:
Only when switching the kube-proxy mode and a previous mode was set. After the next reconcile cleanup is set to false and init container will be removed from the ds.

Reply:
The next reconciliation might happen only in the next maintenance time window. Intermittent restarts will lead to service disruptions as @vpnachev said.